### ISSUE TITLE
Question about the dataset

### ISSUE THREAD

**Issue by Morning-Z** [watreyoung/RACG#6]
Just to confirm, is the dataset for your LLM experiments the one you linked in the paper?

**Comment by watreyoung** [comment]
Sure, we did all the experiments on the dataset linked in our paper, so that the results can be compared and discussed. 

In fact, you can run the code in [Retriever](https://github.com/watreyoung/RACG/tree/main/Retriever) to get your own retrieval-augmented datasets, not only CONCODE, CoNaLa and HearthStone.

**Comment by Morning-Z** [comment]
Why is it that after running `run_rag_pattern.py`, I'm seeing the following behavior:

For both the **CoNaLa** and **HearthStone** datasets, all entries have the exact same five `retrieved_nl `and `retrieved_code `values.

The retrieved_code for the test set entries cannot be found in the training data.

**Comment by watreyoung** [comment]
I got your point.
To help LLM understand the retrieved examples, it is required to provide NL-code pairs in `run_rag_pattern.py`. The file format is a little different for PLMs and LLMs.

I'm sure that you can try to match and find the corresponding NL from the training files based on the linked datasets.

**Comment by Morning-Z** [comment]
I have another question about the data. Why are there identical entries in the retrieval datasets? For instance, the test set for **CoNaLa** (using the `cocosoda` method) contains several exact duplicates.

**Comment by watreyoung** [comment]
I think this problem you mentioned may be caused by the **CoNaLa** dataset it self. There are some instances that have same nl description but different code snippets in this dataset. 

I just checked our retrieval-augmented dataset by `cocosoda` and I did not find the exact duplicated instances. Some code snippets may be retrieved frequently, but it is decided by the retrieval models. Our experimental results have demonstrated the power of `cocosoda` as the SOTA retrieval models of the past.

**Comment by Morning-Z** [comment]
First, for the **CoNaLa** and **Hearthstone** datasets, is the retrieval performed on the training set for both? I am asking because I cannot find the corresponding `retrieved_code` in their respective training sets.

Second, the image shows my reproduced results for **Table 6** based on the code and dataset you provided. There are some discrepancies compared to your results. Could you help me analyze why this might be the case?

<img width="607" height="398" alt="Image" src="https://github.com/user-attachments/assets/488a9565-7e9e-4130-9e82-d70d53c75b93" />
